### PR TITLE
Remove canonicalization of syntax tree JSON

### LINKED ISSUE #32204: OOM when we try to compile module with 1000 classes
**Description:**
I have a module with 1000 class definitions as below.

```
class Person1 {
    string name;
    int age;
    function init(string name, int age) {
        self.name = name;
        self.age = age;
    }
    function getAge() returns int {
        return self.age;

    }
}

class Person2 {
    string name;
    int age;
    function init(string name, int age) {
        self.name = name;
        self.age = age;
    }
    function getAge() returns int {
        return self.age;

    }
}

.
.
.
class Person1000 {
    string name;
    int age;
    function init(string name, int age) {
        self.name = name;
        self.age = age;
    }
    function getAge() returns int {
        return self.age;

    }
}
```
Getting OOM when try to compile the program.
```
[2021-08-17 22:35:07,980] SEVERE {b7a.log.crash} - Java heap space 
java.lang.OutOfMemoryError: Java heap space
        at java.base/java.util.Arrays.copyOf(Arrays.java:3745)
        at java.base/java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:172)
        at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:748)
        at java.base/java.lang.StringBuilder.append(StringBuilder.java:241)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serializeString(JsonCanonicalizer.java:92)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:127)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:110)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:121)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:110)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:110)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:121)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:110)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:121)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.serialize(JsonCanonicalizer.java:110)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.<init>(JsonCanonicalizer.java:37)
        at org.ballerinalang.observability.anaylze.JsonCanonicalizer.getEncodedString(JsonCanonicalizer.java:45)
        at org.ballerinalang.observability.anaylze.DefaultObservabilitySymbolCollector.generateCanonicalJsonString(DefaultObservabilitySymbolCollector.java:177)
        at org.ballerinalang.observability.anaylze.DefaultObservabilitySymbolCollector.writeToExecutable(DefaultObservabilitySymbolCollector.java:114)
        at io.ballerina.projects.JarResolver.getJarFilePathsRequiredForExecution(JarResolver.java:105)
        at io.ballerina.projects.JBallerinaBackend.emitExecutable(JBallerinaBackend.java:510)
        at io.ballerina.projects.JBallerinaBackend.emit(JBallerinaBackend.java:202)
        at io.ballerina.cli.task.CreateExecutableTask.execute(CreateExecutableTask.java:97)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:330)
        at io.ballerina.cli.launcher.Main$$Lambda$22/0x00000001000bc040.accept(Unknown Source)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:53)
```

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #32204

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by warunalakshitha — APPROVED** [review]
LGTM

File: tests/jballerina-unit-test/src/test/resources/test-src/syntaxtree/modules/moduleA/main.bal [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    public function init(int id, string name, string qualification) {
+        self.id = id;
+        self.name = name;
+        self.qualification = qualification;
+    }
+
+    public function getName() returns string {
+        return self.name;
+
```
Comment (Nadeeshan96) [inline tests/jballerina-unit-test/src/test/resources/test-src/syntaxtree/modules/moduleA/main.bal:32]:
```suggestion
```

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java [hunk 2]
```diff
@@ -302,18 +307,35 @@ public static ExitDetails run(CompileResult compileResult, List<String> javaOpts
...
+        }
+        actualArgs.add(index++, "-cp");
+
+        String classPathString = System.getProperty("java.class.path") + classPath;
+        // Create an argument file for windows to mitigate the long classpath issue.
+        if (isWindows) {
+            String classPathArgs = "classPathArgs";
+            File classPathArgsFile;
+            try {
+                classPathArgsFile = File.createTempFile(classPathArgs, ".txt");
```
Comment (Nadeeshan96) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java:324]:
```suggestion
            try {
                File classPathArgsFile = File.createTempFile(classPathArgs, ".txt");
```

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java [hunk 3]
```diff
@@ -302,18 +307,35 @@ public static ExitDetails run(CompileResult compileResult, List<String> javaOpts
...
+        final List<String> actualArgs = new ArrayList<>();
+        int index = 0;
+        actualArgs.add(index++, "java");
+        for (String javaOpt : javaOpts) {
+            actualArgs.add(index++, javaOpt);
+        }
+        actualArgs.add(index++, "-cp");
+
+        String classPathString = System.getProperty("java.class.path") + classPath;
+        // Create an argument file for windows to mitigate the long classpath issue.
```
Comment (Nadeeshan96) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java:319]:
```suggestion
        // Create an argument file for the Windows OS to mitigate the long classpath issue.
```

Comment (gabilang) [reply tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java:319]: Isn't it informative enough to use `Windows` here?

Comment (Nadeeshan96) [reply tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java]: It is enough.

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**